design/cryptography.md

```diff
@@ -28,21 +28,22 @@ An additional complexity is that Obscuro has the requirement to support temporar
 - Generation happens when the central sequencer bootstraps the network
 - The Master Seed is shared across Validators using the process described in the [whitepaper](https://whitepaper.obscu.ro/obscuro-whitepaper/amalgamated.html#sharing-the-master-seed)
 
-## 2. Ensure Enclave Encrypted Communication
-- Client-Enclave communication is encrypted using the Obscuro Key and the Client Key
-- Clients encrypt using the Obscuro Public Key
-- Enclave responds encrypting with the Client Public Key ( also known as Viewing Keys )
-- The Obscuro Key is a Key-Pair deterministically calculated from the Master Seed + Genesis Rollup
-- The Public key will be published to the Management contract and used by all obscuro tooling ( like the wallet extension ) to encrypt transactions
-- Enclaves will determine the Private key when calculating the Master Seed + Genesis Rollup 
-  - Enclaves have the Master Seed through the attestation process
-  - Enclaves fetch Genesis Rollup through the Layer 1 blocks
+## 2. Client-Enclave Encrypted Communication
+- Client-Enclave communication is encrypted using both Obscuro Keys and the Client Keys
+- Obscuro Key and Client (Viewing Key) are both asymmetric key pairs
+  - The Private Obscuro Key is deterministically calculated from Master Seed + Genesis Rollup
+  - The Private Client Key is created by the client in any way it chooses
+  - Both the Public keys of Obscuro and Client are derived, via a one way function, from their corresponding Private keys
+- Clients encrypt communication payload using the Public Key of the Obscuro Key ( only holders of the Private Obscuro Key can decrypt this payload )
+- Enclaves respond to Client by encrypting with the Public Client Key ( only holders of the Private Client Key can decrypt this payload )
+- The Public Obscuro Key is published in the Management contract and used by all obscuro tooling ( like the wallet extension )
 
 
 ## 3. Transaction Encryption per Rollup with Revelation Period
 - Key derivation is a process that takes a key and derives a new key' given a set of inputs. 
-  - The new key' does not release any information for old key (one way function)
-  - Given the key and the input the new key' is deterministically calculated/derived
+  - The new key' does not release any information pertaining the old key (one way function)
+  - Given the same key and the same input, the new key' is deterministically calculated/derived
+- Key derivation allows to segregate vulnerability impact. If one key is compromised, other keys, including the original key are not affected
 - To avoid reusing the same key, transaction encryption keys are derived twice
```

Comment:
this is not correct.
The WP talks about how this should be done: https://whitepaper.obscu.ro/obscuro-whitepaper/detailed-design.html#transaction-encryption

Reply:
Good catch. 
Instead of Master Seed -> Rollup number X Encryption Key -> 5 Encryption Keys. We should have Master Seed -> 5 Encryption Keys.

Updated 👍